src/less/chart-legend/chart-legend.less

```diff
@@ -0,0 +1,85 @@
+.chart-legend {
+    width: 100%;
+}
+
+.chart-legend dl {
+    display: flex;
+    flex-direction: column;
+    margin: 0;
+    padding: 0;
+    row-gap: var(--spacing-200);
+}
+
+.chart-legend__item {
+    align-items: center;
+    display: flex;
+    font-size: var(--font-size-small);
+    width: 100%;
+
+    &::before {
+        border: 1px solid;
+        border-radius: 2px;
+        box-sizing: border-box;
+        content: "";
+        display: inline-block;
+        height: 10px;
+        margin-inline-end: var(--spacing-100);
+        width: 10px;
+    }
+
+    &.chart-legend__item--primary {
+        &::before {
+            background-color: var(--color-data-viz-chart-primary);
+            border-color: var(--color-data-viz-chart-primary);
+        }
+    }
+
+    &.chart-legend__item--secondary {
+        &::before {
+            background-color: var(--color-data-viz-chart-secondary);
+            border-color: var(--color-data-viz-chart-secondary);
+        }
+    }
+
+    &.chart-legend__item--tertiary {
+        &::before {
+            background: repeating-linear-gradient(
+                30deg,
+                var(--color-data-viz-chart-tertiary-background),
+                var(--color-data-viz-chart-tertiary-background) 2px,
+                var(--color-data-viz-chart-tertiary-stroke) 2px,
+                var(--color-data-viz-chart-tertiary-stroke) 3px,
+                var(--color-data-viz-chart-tertiary-background) 3px,
+                var(--color-data-viz-chart-tertiary-background) 4px
+            );
+            border: 1px solid var(--color-data-viz-chart-tertiary-stroke);
+        }
+    }
+
+    &.chart-legend__item--quaternary {
+        &::before {
+            background: repeating-linear-gradient(
+                0deg,
+                var(--color-data-viz-chart-quaternary-background),
+                var(--color-data-viz-chart-quaternary-background) 2px,
+                var(--color-data-viz-chart-quaternary-stroke) 2px,
+                var(--color-data-viz-chart-quaternary-stroke) 3px,
+                var(--color-data-viz-chart-quaternary-background) 3px,
+                var(--color-data-viz-chart-quaternary-background) 4px
+            );
+            border: 1px solid var(--color-data-viz-chart-quaternary-stroke);
+        }
+    }
+
+    &.chart-legend__item--quinary {
+        &::before {
+            background-color: var(--color-data-viz-chart-quinary-background);
+            border-color: var(--color-data-viz-chart-quinary-stroke);
+        }
+    }
+}
+
+.chart-legend__value {
+    flex-grow: 1;
+    text-align: right;
```

Comment:
If we end up keeping it this way (and I hope we don't), I believe we'd need an `RTL` treatment for this as well.